// @thumbtack/thumbprint-react/dist/cjs/components/InputRow/index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var warning = _interopDefault(require('warning'));
var styles = _interopDefault(require('./index.module.scss'));

// will then be able to know when their component is being used within an `InputRow` as well as
// whether it is the first of last child.

var InputRowContext = React__default.createContext({
  isWithinInputRow: false,
  isFirstInputRowChild: false,
  isLastInputRowChild: false
});
function InputRow(_ref) {
  var widthRatios = _ref.widthRatios,
      children = _ref.children,
      dataTest = _ref.dataTest,
      dataTestId = _ref.dataTestId;
  var widthLength = widthRatios && widthRatios.length;
  var numChildren = React__default.Children.count(children);
  warning(!widthLength || widthLength === numChildren, "The length of prop `width` (".concat(widthLength, ") must match the number of children (").concat(numChildren, ") passed to `InputRow`."));
  warning(numChildren > 1, "The `InputRow` component expects more than one child within the `children` prop. ".concat(numChildren, " received."));
  return React__default.createElement("div", {
    className: styles.root,
    "data-test": dataTest,
    "data-testid": dataTestId
  }, React__default.Children.map(children, function (child, i) {
    return React__default.createElement("div", {
      style: {
        flex: widthRatios ? "".concat(widthRatios[i]) : undefined
      }
    }, React__default.createElement(InputRowContext.Provider, {
      // eslint-disable-next-line react/jsx-no-constructed-context-values
      value: {
        isWithinInputRow: true,
        isFirstInputRowChild: i === 0,
        isLastInputRowChild: i + 1 === numChildren
      }
    }, child));
  }));
}

exports.InputRowContext = InputRowContext;
exports.default = InputRow;
