// @thumbtack/thumbprint-react/dist/cjs/components/ModalCurtain/index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var React__default = _interopDefault(React);
var classNames = _interopDefault(require('classnames'));
var noScroll = require('./components/no-scroll.js');
var useCloseOnEscape = require('../../utils/use-close-on-escape.js');
var useFocusTrap = require('../../utils/use-focus-trap.js');
var ConditionalPortal = require('../../utils/ConditionalPortal.js');
var styles = _interopDefault(require('./index.module.scss'));

function ModalCurtain(_ref) {
  var _classNames;

  var _ref$stage = _ref.stage,
      stage = _ref$stage === void 0 ? 'exited' : _ref$stage,
      _ref$shouldCloseOnEsc = _ref.shouldCloseOnEscape,
      shouldCloseOnEscape = _ref$shouldCloseOnEsc === void 0 ? true : _ref$shouldCloseOnEsc,
      _ref$accessibilityLab = _ref.accessibilityLabel,
      accessibilityLabel = _ref$accessibilityLab === void 0 ? 'Modal' : _ref$accessibilityLab,
      initialFocusProp = _ref.initialFocus,
      onCloseClick = _ref.onCloseClick,
      children = _ref.children;

  var _useState = React.useState(false),
      _useState2 = _rollupPluginBabelHelpers.slicedToArray(_useState, 2),
      isClient = _useState2[0],
      setIsClient = _useState2[1];

  var _useState3 = React.useState(null),
      _useState4 = _rollupPluginBabelHelpers.slicedToArray(_useState3, 2),
      wrapperEl = _useState4[0],
      setWrapperEl = _useState4[1];

  React.useEffect(function () {
    setIsClient(true);
  }, []);
  var initialFocus = initialFocusProp || wrapperEl;
  var isEnteringOrEntered = stage === 'entering' || stage === 'entered';
  var shouldBindEscListener = isClient && shouldCloseOnEscape && isEnteringOrEntered;
  var shouldTrapFocus = isClient && !!initialFocus && stage === 'entered';
  var shouldDisableScrolling = isEnteringOrEntered;
  useCloseOnEscape.default(onCloseClick, shouldBindEscListener);
  useFocusTrap.default(wrapperEl, shouldTrapFocus, initialFocus);
  return React__default.createElement(ConditionalPortal.default, {
    shouldDisplace: true
  }, React__default.createElement("div", {
    role: "dialog",
    "aria-label": accessibilityLabel,
    tabIndex: -1,
    ref: function ref(element) {
      setWrapperEl(element);
    }
  }, shouldDisableScrolling && React__default.createElement(noScroll.default, null), children && children({
    curtainOnClick: function curtainOnClick(event) {
      // Ensures that the click event happened on the element that has the
      // `onClick`. This prevents clicks deep within `children` from bubbling
      // up and closing the ModalCurtain.
      if (event.target === event.currentTarget) {
        onCloseClick();
      }
    },
    curtainClassName: classNames((_classNames = {}, _rollupPluginBabelHelpers.defineProperty(_classNames, styles.root, true), _rollupPluginBabelHelpers.defineProperty(_classNames, styles.rootOpen, isEnteringOrEntered), _classNames))
  })));
}

exports.default = ModalCurtain;
