// @thumbtack/thumbprint-tokens/dist/javascript-cjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tpColorRed600 = exports.tpColorRed500 = exports.tpColorRed = exports.tpColorRed300 = exports.tpColorRed200 = exports.tpColorRed100 = exports.tpColorYellow600 = exports.tpColorYellow500 = exports.tpColorYellow = exports.tpColorYellow300 = exports.tpColorYellow200 = exports.tpColorYellow100 = exports.tpColorGreen600 = exports.tpColorGreen500 = exports.tpColorGreen = exports.tpColorGreen300 = exports.tpColorGreen200 = exports.tpColorGreen100 = exports.tpColorPurple600 = exports.tpColorPurple500 = exports.tpColorPurple = exports.tpColorPurple300 = exports.tpColorPurple200 = exports.tpColorPurple100 = exports.tpColorIndigo600 = exports.tpColorIndigo500 = exports.tpColorIndigo = exports.tpColorIndigo300 = exports.tpColorIndigo200 = exports.tpColorIndigo100 = exports.tpColorBlue600 = exports.tpColorBlue500 = exports.tpColorBlue = exports.tpColorBlue300 = exports.tpColorBlue200 = exports.tpColorBlue100 = exports.tpBreakpointSplitViewSmallValue = exports.tpBreakpointSplitViewSmall = exports.tpBreakpointSplitViewMediumValue = exports.tpBreakpointSplitViewMedium = exports.tpBreakpointSmallValue = exports.tpBreakpointSmall = exports.tpBreakpointMediumValue = exports.tpBreakpointMedium = exports.tpBreakpointLargeValue = exports.tpBreakpointLarge = exports.tpBorderRadiusSides = exports.tpBorderRadiusFull = exports.tpBorderRadiusBig = exports.tpBorderRadiusBase = void 0;
exports.tpFontTitle6Weight = exports.tpFontTitle6LineHeight = exports.tpFontTitle6Size = exports.tpFontTitle5Weight = exports.tpFontTitle5LineHeight = exports.tpFontTitle5Size = exports.tpFontTitle4Weight = exports.tpFontTitle4LineHeight = exports.tpFontTitle4Size = exports.tpFontTitle3ResponsiveWeight = exports.tpFontTitle3ResponsiveLineHeight = exports.tpFontTitle3ResponsiveSize = exports.tpFontTitle3Weight = exports.tpFontTitle3LineHeight = exports.tpFontTitle3Size = exports.tpFontTitle2ResponsiveWeight = exports.tpFontTitle2ResponsiveLineHeight = exports.tpFontTitle2ResponsiveSize = exports.tpFontTitle2Weight = exports.tpFontTitle2LineHeight = exports.tpFontTitle2Size = exports.tpFontTitle1ResponsiveWeight = exports.tpFontTitle1ResponsiveLineHeight = exports.tpFontTitle1ResponsiveSize = exports.tpFontTitle1Weight = exports.tpFontTitle1LineHeight = exports.tpFontTitle1Size = exports.tpFontWeightBold = exports.tpFontWeightNormal = exports.tpFontFamilyMonospace = exports.tpFontFamilyBase = exports.tpEaseInOut = exports.tpEaseOut = exports.tpEaseIn = exports.tpDuration6 = exports.tpDuration5 = exports.tpDuration4 = exports.tpDuration3 = exports.tpDuration2 = exports.tpDuration1 = exports.tpCornerRadiusSides = exports.tpCornerRadiusFull = exports.tpCornerRadiusBig = exports.tpCornerRadiusBase = exports.tpColorWhite = exports.tpColorGray = exports.tpColorGray300 = exports.tpColorGray200 = exports.tpColorBlack = exports.tpColorBlack300 = void 0;
exports.tpZIndexModal = exports.tpWrapNoPadWidth = exports.tpWrapMaxWidth = exports.tpSpace8 = exports.tpSpace7 = exports.tpSpace6 = exports.tpSpace5 = exports.tpSpace4 = exports.tpSpace3 = exports.tpSpace2 = exports.tpSpace1 = exports.tpScrimDark80 = exports.tpScrimLight80 = exports.tpShadowCard = exports.tpShadow400 = exports.tpShadow300 = exports.tpShadow200 = exports.tpShadow100 = exports.tpLineHeightLoose = exports.tpLineHeightTight = exports.tpLineHeightBase = exports.tpLetterSpacingExtraTight = exports.tpLetterSpacingTight = exports.tpLetterSpacingLoose = exports.tpFontBody3LineHeight = exports.tpFontBody3Size = exports.tpFontBody2LineHeight = exports.tpFontBody2Size = exports.tpFontBody1LineHeight = exports.tpFontBody1Size = exports.tpFontTitle8Weight = exports.tpFontTitle8LineHeight = exports.tpFontTitle8Size = exports.tpFontTitle7Weight = exports.tpFontTitle7LineHeight = exports.tpFontTitle7Size = void 0;
// @deprecated
exports.tpBorderRadiusBase = "4px";
// @deprecated
exports.tpBorderRadiusBig = "6px";
// @deprecated
exports.tpBorderRadiusFull = "50%";
// @deprecated
exports.tpBorderRadiusSides = "9999px";
exports.tpBreakpointLarge = "1025px";
exports.tpBreakpointLargeValue = 1025;
exports.tpBreakpointMedium = "700px";
exports.tpBreakpointMediumValue = 700;
exports.tpBreakpointSmall = "481px";
exports.tpBreakpointSmallValue = 481;
exports.tpBreakpointSplitViewMedium = "1060px";
exports.tpBreakpointSplitViewMediumValue = 1060;
exports.tpBreakpointSplitViewSmall = "769px";
exports.tpBreakpointSplitViewSmallValue = 769;
exports.tpColorBlue100 = "#eaf6fa";
exports.tpColorBlue200 = "#b3ebff";
exports.tpColorBlue300 = "#79d2f2";
exports.tpColorBlue = "#009fd9";
exports.tpColorBlue500 = "#007fad";
exports.tpColorBlue600 = "#005979";
exports.tpColorIndigo100 = "#e8f1fd";
exports.tpColorIndigo200 = "#cce1ff";
exports.tpColorIndigo300 = "#96c2ff";
exports.tpColorIndigo = "#5968e2";
exports.tpColorIndigo500 = "#4f54b3";
exports.tpColorIndigo600 = "#383c80";
exports.tpColorPurple100 = "#f5efff";
exports.tpColorPurple200 = "#dfccff";
exports.tpColorPurple300 = "#c9acfd";
exports.tpColorPurple = "#8d56eb";
exports.tpColorPurple500 = "#6637b6";
exports.tpColorPurple600 = "#492782";
exports.tpColorGreen100 = "#e1fdf3";
exports.tpColorGreen200 = "#c6f7da";
exports.tpColorGreen300 = "#73e4a2";
exports.tpColorGreen = "#2db783";
exports.tpColorGreen500 = "#16855b";
exports.tpColorGreen600 = "#054e33";
exports.tpColorYellow100 = "#fdf7e7";
exports.tpColorYellow200 = "#ffebb3";
exports.tpColorYellow300 = "#ffdd80";
exports.tpColorYellow = "#febe14";
exports.tpColorYellow500 = "#a77005";
exports.tpColorYellow600 = "#714601";
exports.tpColorRed100 = "#ffeff0";
exports.tpColorRed200 = "#ffd9d9";
exports.tpColorRed300 = "#ffb0b0";
exports.tpColorRed = "#ff5a5f";
exports.tpColorRed500 = "#b22d31";
exports.tpColorRed600 = "#7d0d10";
exports.tpColorBlack300 = "#676d73";
exports.tpColorBlack = "#2f3033";
exports.tpColorGray200 = "#fafafa";
exports.tpColorGray300 = "#e9eced";
exports.tpColorGray = "#d3d4d5";
exports.tpColorWhite = "#ffffff";
exports.tpCornerRadiusBase = "4px";
exports.tpCornerRadiusBig = "6px";
exports.tpCornerRadiusFull = "50%";
exports.tpCornerRadiusSides = "9999px";
exports.tpDuration1 = 75;
exports.tpDuration2 = 150;
exports.tpDuration3 = 200;
exports.tpDuration4 = 250;
exports.tpDuration5 = 300;
exports.tpDuration6 = 350;
exports.tpEaseIn = "cubic-bezier(0.50, 0, 1, 1)";
exports.tpEaseOut = "cubic-bezier(0, 0, 0.40, 1)";
exports.tpEaseInOut = "cubic-bezier(0.45, 0, 0.40, 1)";
exports.tpFontFamilyBase = "Mark, Avenir, Helvetica, Arial, sans-serif";
exports.tpFontFamilyMonospace = "'Source Code Pro', monospace";
exports.tpFontWeightNormal = "400";
exports.tpFontWeightBold = "700";
exports.tpFontTitle1Size = "28px";
exports.tpFontTitle1LineHeight = "32px";
exports.tpFontTitle1Weight = "700";
exports.tpFontTitle1ResponsiveSize = "40px";
exports.tpFontTitle1ResponsiveLineHeight = "44px";
exports.tpFontTitle1ResponsiveWeight = "700";
exports.tpFontTitle2Size = "24px";
exports.tpFontTitle2LineHeight = "28px";
exports.tpFontTitle2Weight = "700";
exports.tpFontTitle2ResponsiveSize = "32px";
exports.tpFontTitle2ResponsiveLineHeight = "40px";
exports.tpFontTitle2ResponsiveWeight = "700";
exports.tpFontTitle3Size = "22px";
exports.tpFontTitle3LineHeight = "28px";
exports.tpFontTitle3Weight = "700";
exports.tpFontTitle3ResponsiveSize = "24px";
exports.tpFontTitle3ResponsiveLineHeight = "32px";
exports.tpFontTitle3ResponsiveWeight = "700";
exports.tpFontTitle4Size = "20px";
exports.tpFontTitle4LineHeight = "28px";
exports.tpFontTitle4Weight = "700";
exports.tpFontTitle5Size = "18px";
exports.tpFontTitle5LineHeight = "24px";
exports.tpFontTitle5Weight = "700";
exports.tpFontTitle6Size = "16px";
exports.tpFontTitle6LineHeight = "24px";
exports.tpFontTitle6Weight = "700";
exports.tpFontTitle7Size = "14px";
exports.tpFontTitle7LineHeight = "20px";
exports.tpFontTitle7Weight = "700";
exports.tpFontTitle8Size = "12px";
exports.tpFontTitle8LineHeight = "18px";
exports.tpFontTitle8Weight = "700";
exports.tpFontBody1Size = "16px";
exports.tpFontBody1LineHeight = "24px";
exports.tpFontBody2Size = "14px";
exports.tpFontBody2LineHeight = "20px";
exports.tpFontBody3Size = "12px";
exports.tpFontBody3LineHeight = "18px";
exports.tpLetterSpacingLoose = "1px";
exports.tpLetterSpacingTight = "-1px";
exports.tpLetterSpacingExtraTight = "-2px";
exports.tpLineHeightBase = "1.6";
exports.tpLineHeightTight = "1.4";
exports.tpLineHeightLoose = "1.9";
exports.tpShadow100 = "0px 1px 3px rgba(0, 0, 0, 0.1)";
exports.tpShadow200 = "0px 2px 4px rgba(0, 0, 0, 0.15)";
exports.tpShadow300 = "0px 2px 7px rgba(0, 0, 0, 0.15)";
exports.tpShadow400 = "0px 2px 10px rgba(0, 0, 0, 0.2)";
// @deprecated
exports.tpShadowCard = "0 -1px 1px 0 rgba(0, 0, 0, 0.04), 0 1px 2px 0 rgba(0, 0, 0, 0.16)";
exports.tpScrimLight80 = "rgba(255, 255, 255, 0.8)";
exports.tpScrimDark80 = "rgba(0, 0, 0, 0.8)";
exports.tpSpace1 = "4px";
exports.tpSpace2 = "8px";
exports.tpSpace3 = "16px";
exports.tpSpace4 = "24px";
exports.tpSpace5 = "32px";
exports.tpSpace6 = "64px";
exports.tpSpace7 = "128px";
exports.tpSpace8 = "256px";
exports.tpWrapMaxWidth = "946px";
exports.tpWrapNoPadWidth = "1010px";
exports.tpZIndexModal = 200;
